Source/Core/AST/Absy.cs

```diff
@@ -280,6 +280,33 @@ public List<int> FindLayers()
       }
       return layers.Distinct().OrderBy(l => l).ToList();
     }
+
+    // Look for {:name string} in list of attributes.
+    public string FindStringAttribute(string name)
```

Comment:
If you define these methods as extension methods, then you can also call them without casting to `ICarriesAttributes` first.